views/pages/fixture_editor_vue.js

```diff
@@ -0,0 +1,659 @@
+  str += '<input type="text" required pattern="[a-z0-9\-]+" title="Use only lowercase letters, numbers and dashes." v-model="fixture.newManufacturerShortName" />';
+  str += '</label>';
+  str += '</section>';
+
+  str += '<section class="new-manufacturer-website">';
+  str += '<label>';
+  str += '<span class="label">Website</span>';
+  str += urlInput('fixture.newManufacturerWebsite', properties.manufacturer.website);
+  str += '</label>';
+  str += '</section>';
+
+  str += '<section class="new-manufacturer-comment">';
+  str += '<label>';
+  str += '<span class="label">Comment</span>';
+  str += textInput('fixture.newManufacturerComment', properties.manufacturer.comment);
+  str += '</label>';
+  str += '</section>';
+  str += '<div class="button-bar">or <a href="#use-existing-manufacturer" @click.prevent="existingManufacturer">choose an existing manufacturer</a></div>';
+  str += '</section>'; // [v-else]
+
+  str += '</section>'; // .manufacturer
+
+
+  // Fixture info
+  str += '<section class="fixture-info card">';
+  str += '<h2>Fixture info</h2>';
+
+  str += '<section class="fixture-name">';
+  str += '<label>';
+  str += '<span class="label">Name</span>';
+  str += textInput('fixture.name', properties.fixture.name);
+  str += '</label>';
+  str += '</section>';
+
+  str += '<section class="fixture-shortName">';
+  str += '<label>';
+  str += '<span class="label">Unique short name</span>';
+  str += textInput('fixture.shortName', properties.fixture.shortName, 'defaults to name');
+  str += '</label>';
+  str += '</section>';
+
+  str += '<section class="categories">';
+  str += '<label>';
+  str += '<span class="label">Categories</span>';
+  str += `<select multiple required size="${Object.keys(options.register.categories).length}" v-model="fixture.categories">`;
+  for (const cat of properties.category.enum) {
+    str += `<option value="${cat}">${cat}</option>`;
+  }
+  str += '</select></label>';
+  str += '</section>';
+
+  str += '<section class="comment">';
+  str += '<label>';
+  str += '<span class="label">Comment</span>';
+  str += textareaInput('fixture.comment', properties.fixture.comment);
+  str += '</label>';
+  str += '</section>';
+
+  str += '<section class="manualURL">';
+  str += '<label>';
+  str += '<span class="label">Manual URL</span>';
+  str += urlInput('fixture.manualURL', properties.fixture.manualURL);
+  str += '</label>';
+  str += '</section>';
+
+  str += '</section>'; // .fixture-info
+
+
+  // Fixture physical
+  str += '<section class="physical card">';
+  str += '<h2>Physical data</h2>';
+  str += '<physical-data v-model="fixture.physical"></physical-data>';
+  str += '</section>';
+
+
+  // Fixture modes
+  str += '<section class="fixture-modes">';
+  str += '<fixture-mode v-for="(mode, index) in fixture.modes" :mode="mode" :fixture="fixture" :channel="channel" :key="mode.uuid" @remove="fixture.modes.splice(index, 1)" @open-channel-dialog="openChannelDialog"></fixture-mode>';
```

Comment:
It might be considerable breaking such lines down into several lines, like that:

```js
  str += '<fixture-mode'
      +  '  v-for="(mode, index) in fixture.modes"'
      +  '  :mode="mode"'
      +  '  :fixture="fixture"'
      +  '  :channel="channel"'
      +  '  :key="mode.uuid"'
      +  '  @remove="fixture.modes.splice(index, 1)"'
      +  '  @open-channel-dialog="openChannelDialog">';
  str += '</fixture-mode>';
```

This improves readability a lot.